### ISSUE TITLE
Feature Suggestion - Camo Categories View

### ISSUE THREAD

**Issue by sol3uk** [carlssonemil/orion#3]
Thanks for following up with another app like this, really love the formula you've got with all the trackers, they work really well!

So this game mixes things up a fair bit. It's far more focused on individual camos, allowing you to unlock them across all guns. With that in mind I have a new use-case that I think many other people will also be in the same boat as me, in wanting to track specific camos and categories of camos. Including the _**level I need the weapon to be**_ to unlock the challenge, as well as **_the actual challenge itself_**.
There are 15 Camo Categories (Including a Mastery Category):
- Spray Paint
- Woodland
- Digital
- Dragon
- Geometric
- Fun
- Foliage
- Skulls
- Tiger
- Stripes
- Reptile
- Solid Colours
- Classic
- Cliffside
- Completionist

My suggestion would be that you have an alternative view of the camos, where the camos are the stars of the show (not the guns), and you can pin/star/favourite the camos you are most interested in so you can bee-line those.
In the screenshot you can see the most important info that the game buries in it's menus, this is what I would expect to be front and centre in a tracking app that focuses on the camos. (whilst also showing the eventual challenge you will need to complete as well as the level requirement)

![image](https://user-images.githubusercontent.com/26093813/199630074-da993629-ab8b-41c6-b131-e4a2e4d93e78.png)

There could be a similar design/layout, but the camos are the stars. Here's a (really bad and quick) mock-up: 

![image](https://user-images.githubusercontent.com/26093813/199631721-432bf9f1-3b7d-4591-89b4-8d8dd2059635.png)

Let me know what you think of the idea and how you'd implement it if at all, and I can get to work on a PR with something like this to expand the functionality of the app for people more like me, bee-lining specific fun camos 😄

**Comment by sol3uk** [comment]
I've had a crack at implementing the basics behind this, one thing I noticed is that because the weapons are so core to the application, adding camos in the same way weapons exist would cause an inadvertent issue where any camo completion from this view wouldn't carry over without a bit of a hacky solution to retroactively update the camo progress on the weapon. 
I'll have another go at building a design into the existing weapon model whilst trying to keep it simple.

**Comment by sol3uk** [comment]
Here's what I was thinking, for context (there's definitely a better way to do this):

![image](https://user-images.githubusercontent.com/26093813/199736437-900c814f-42b7-42bf-b332-5cb64ed51da2.png)

The main idea here is to include the weapon level requirement as useable information, whether or not it should be "trackable" is debateable, at the moment I'm leaning towards it not needing to be trackable, just visually available.

**Comment by carlssonemil** [comment]
Great job with doing part 1 of this feature with the data structure in #4!

Do you want me to take a shot at implementing this feature in the UI, or try your hand at it yourself? 😃

**Comment by sol3uk** [comment]
I can give it a go definitely, I've already got the skeleton of the camo components ready to go, just based off the weapon one, but I feel like there will some bits I'll miss or won't be able to make look great 😅

**Comment by sol3uk** [comment]
If you want to have a go at it today tho have at it and let me know the branch! I'll be busy most of today so I won't get something together until probably next week.

**Comment by carlssonemil** [comment]
No rush, do it when you have the time! I'm busy today as well. I might, maybe, take a look at it tomorrow, if I'm not looking at other side projects or am busy gaming 👀 

But feel free to whip something up and I can come with suggestions if you're unsure of the design-part! 😄 I appreciate any help with this project!

**Comment by mrsloth4994** [comment]
hey dude, when will you update the remain camos, polyatomic/orion?

**Comment by carlssonemil** [comment]
> hey dude, when will you update the remain camos, polyatomic/orion?

Do you mean the images? There's about 181 camouflages in this game, it'll take a few weeks to gather them all my friend 😄

**Comment by mrsloth4994** [comment]
I mean the polyatomic and Orion camo like you've done for base camo, gold, platinum

**Comment by carlssonemil** [comment]
> I mean the polyatomic and Orion camo like you've done for base camo, gold, platinum

Once you unlock Platinum on every gun, the Polyatomic will be visible, and Orion is unlocked automatically once you have Polyatomic on 51 weapons 😄

**Comment by mrsloth4994** [comment]
I mean the icon bro, like you have green for base camos, then gold, platinum etc. Are you not putting them colours on just for show🤷‍♂️ think it'll look pretty dope if you did

**Comment by sol3uk** [comment]
@mrsloth4994 it does appear on the app, just not until you've checked off all camos:
![image](https://user-images.githubusercontent.com/26093813/200400719-3ec27974-ee2a-4056-ab7a-07fbed57b930.png)

**Comment by carlssonemil** [comment]
> @mrsloth4994 it does appear on the app, just not until you've checked off all camos: ![image](https://user-images.githubusercontent.com/26093813/200400719-3ec27974-ee2a-4056-ab7a-07fbed57b930.png)

I need to set a max-height on the Polyatomic though 😅 @mrsloth4994, the polyatomic shows up once you have completed 51 platinum camouflages. I didn't think it needed to be shown until then as you can't finish it.

**Comment by sol3uk** [comment]
I think it might be worth having it available so the double click functionality can be used for people that have already done part of the grind @carlssonemil?

**Comment by mrsloth4994** [comment]
Orion tho😢

**Comment by sol3uk** [comment]
@mrsloth4994 the orion camo unlocks when you get 51 polyatomic camos. AFAIK there's no way you can unlock it per-gun, so I'm not sure it why you would want to show it under each gun?

**Comment by mrsloth4994** [comment]
Wait so when you get 51 platinum guns you get polyatomic? No challenge requirement after you unlock the camo?

**Comment by carlssonemil** [comment]
> I think it might be worth having it available so the double click functionality can be used for people that have already done part of the grind @carlssonemil?

Hm.. 🤔 I'll think about it.

**Comment by carlssonemil** [comment]
> Wait so when you get 51 platinum guns you get polyatomic? No challenge requirement after you unlock the camo?

https://orion.emca.app/requirements

You need 51 platinum to unlock the Polyatomic challenge, it won't be available in-game until you have 51 platinum. And then you need 51 polyatomic to unlock the Orion camouflage.

**Comment by carlssonemil** [comment]
Fixed in #10
